[Snyk] Upgrade @types/react from 18.2.12 to 18.2.45

techcobweb opened the pull request:
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to upgrade @types/react from 18.2.12 to 18.2.45.</h3>

:information_source: Keep your dependencies up-to-date. This makes it easier to fix existing vulnerabilities and to more quickly identify and fix newly disclosed vulnerabilities when they affect your project.
<hr/>

- The recommended version is **33 versions** ahead of your current version.
- The recommended version was released **a month ago**, on 2023-12-13.


<details>
<summary><b>Release notes</b></summary>
<br/>
  <details>
    <summary>Package name: <b>@types/react</b></summary>
    <ul>
      <li>
        <b>18.2.45</b> - 2023-12-13
      </li>
      <li>
        <b>18.2.44</b> - 2023-12-12
      </li>
      <li>
        <b>18.2.43</b> - 2023-12-09
      </li>
      <li>
        <b>18.2.42</b> - 2023-12-04
      </li>
      <li>
        <b>18.2.41</b> - 2023-12-02
      </li>
      <li>
        <b>18.2.40</b> - 2023-12-01
      </li>
      <li>
        <b>18.2.39</b> - 2023-11-27
      </li>
      <li>
        <b>18.2.38</b> - 2023-11-21
      </li>
      <li>
        <b>18.2.37</b> - 2023-11-07
      </li>
      <li>
        <b>18.2.36</b> - 2023-11-06
      </li>
      <li>
        <b>18.2.35</b> - 2023-11-05
      </li>
      <li>
        <b>18.2.34</b> - 2023-11-01
      </li>
      <li>
        <b>18.2.33</b> - 2023-10-26
      </li>
      <li>
        <b>18.2.32</b> - 2023-10-25
      </li>
      <li>
        <b>18.2.31</b> - 2023-10-20
      </li>
      <li>
        <b>18.2.30</b> - 2023-10-19
      </li>
      <li>
        <b>18.2.29</b> - 2023-10-18
      </li>
      <li>
        <b>18.2.28</b> - 2023-10-10
      </li>
      <li>
        <b>18.2.27</b> - 2023-10-09
      </li>
      <li>
        <b>18.2.26</b> - 2023-10-09
      </li>
      <li>
        <b>18.2.25</b> - 2023-10-04
      </li>
      <li>
        <b>18.2.24</b> - 2023-10-01
      </li>
      <li>
        <b>18.2.23</b> - 2023-09-26
      </li>
      <li>
        <b>18.2.22</b> - 2023-09-18
      </li>
      <li>
        <b>18.2.21</b> - 2023-08-22
      </li>
      <li>
        <b>18.2.20</b> - 2023-08-09
      </li>
      <li>
        <b>18.2.19</b> - 2023-08-08
      </li>
      <li>
        <b>18.2.18</b> - 2023-07-31
      </li>
      <li>
        <b>18.2.17</b> - 2023-07-26
      </li>
      <li>
        <b>18.2.16</b> - 2023-07-24
      </li>
      <li>
        <b>18.2.15</b> - 2023-07-13
      </li>
      <li>
        <b>18.2.14</b> - 2023-06-23
      </li>
      <li>
        <b>18.2.13</b> - 2023-06-19
      </li>
      <li>
        <b>18.2.12</b> - 2023-06-12
      </li>
    </ul>
    from <a href="https://snyk.io/redirect/github/DefinitelyTyped/DefinitelyTyped/releases">@types/react GitHub release notes</a>
  </details>
</details>
<hr/>

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open upgrade PRs.*

For more information:  <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiJkYjBhYTQyYi1mM2Y5LTRmZDItODVlMC1lNDk3OWU1M2Q4MjEiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6ImRiMGFhNDJiLWYzZjktNGZkMi04NWUwLWU0OTc5ZTUzZDgyMSJ9fQ==" width="0" height="0"/>

🧐 [View latest project report](https://app.snyk.io/org/techcobweb/project/e90df747-c285-459d-a45e-38b00c5bd2b6?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🛠 [Adjust upgrade PR settings](https://app.snyk.io/org/techcobweb/project/e90df747-c285-459d-a45e-38b00c5bd2b6/settings/integration?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🔕 [Ignore this dependency or unsubscribe from future upgrade PRs](https://app.snyk.io/org/techcobweb/project/e90df747-c285-459d-a45e-38b00c5bd2b6/settings/integration?pkg&#x3D;@types/react&amp;utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr#auto-dep-upgrades)

<!--- (snyk:metadata:{"prId":"db0aa42b-f3f9-4fd2-85e0-e4979e53d821","prPublicId":"db0aa42b-f3f9-4fd2-85e0-e4979e53d821","dependencies":[{"name":"@types/react","from":"18.2.12","to":"18.2.45"}],"packageManager":"npm","type":"auto","projectUrl":"https://app.snyk.io/org/techcobweb/project/e90df747-c285-459d-a45e-38b00c5bd2b6?utm_source=github&utm_medium=referral&page=upgrade-pr","projectPublicId":"e90df747-c285-459d-a45e-38b00c5bd2b6","env":"prod","prType":"upgrade","vulns":[],"issuesToFix":[],"upgrade":[],"upgradeInfo":{"versionsDiff":33,"publishedDate":"2023-12-13T01:31:45.703Z"},"templateVariants":[],"hasFixes":false,"isMajorUpgrade":false,"isBreakingChange":false,"priorityScoreList":[]}) --->

DianaSlepikaite commented:
rerun tests

DianaSlepikaite commented:
run tests

Michael-Baylis commented:
rerun tests

Michael-Baylis commented on src/app/results-page/rasfilters/testclasses-filter/testclasses-filter.component.ts:
```diff
@@ -0,0 +1,74 @@
+              newBundles.push(testClass.bundle);
+              nextId++;
+            }
+            var nextId = 0;
```
Dont need to do anything with Bundles,  you can remove that code.

Michael-Baylis replied:
Bundles will have its own dropdown

snyk-bot committed:
fix: upgrade @types/react from 18.2.12 to 18.2.45

Snyk has created this PR to upgrade @types/react from 18.2.12 to 18.2.45.

See this package in npm:
https://www.npmjs.com/package/@types/react

See this project in Snyk:
https://app.snyk.io/org/techcobweb/project/e90df747-c285-459d-a45e-38b00c5bd2b6?utm_source=github&utm_medium=referral&page=upgrade-pr
